### PR TITLE
Top page

### PR DESCRIPTION
**Pull request by pauliusanuzas**
Added top page, UI and fixed some backend bugs.

### CODE REVIEW

**Review by modestas-jucevicius — APPROVED** [review]
Why comment out tests? :D

**Comment by pauliusanuzas** [comment]
My fault, commented it out while I tried to fix some bugs and then totally forgot about it. Will fix it asap. :D

File: RoadRangerMobileApp/RoadRangerMobileApp/Presenters/TopPresenter.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        {
+            this.view.Search += new EventHandler<EventArgs>(Search);
+        }
+
+        async Task ExecuteLoadItemsCommand()
+        {
+            if (IsBusy)
+            {
+                return;
+            }
```
Comment (aurisgo) [inline RoadRangerMobileApp/RoadRangerMobileApp/Presenters/TopPresenter.cs]: use lock. Look MyGalleryPresenter as example.

File: RoadRangerMobileApp/RoadRangerMobileApp/ViewModels/MyShopViewModel.cs [hunk 2]
```diff
@@ -4,30 +4,41 @@
 using System;
 using System.Collections.Generic;
 using System.Text;
+using Xamarin.Forms;
+using RoadRangerMobileApp.Views;
+using System.Diagnostics;
 
 namespace RoadRangerMobileApp.ViewModels
 {
     public class MyShopViewModel
```
Comment (aurisgo) [inline RoadRangerMobileApp/RoadRangerMobileApp/ViewModels/MyShopViewModel.cs:13]: In MVP structure, we don't have ViewModel. Rename it something like MyShopPresenter and change its location.

File: RoadRangerBackEnd/Score/HighscoresService.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        public List<User> SortedByScore(List<User> users) //Isrykiuoja User lista pagal scora mazejimo tvarka
+        {
+            return users.OrderByDescending(o=>o.Score).ToList();
+        }
+
+        public List<User> Top10(List<User> users)
+        {
+            List<User> sortedUsers = SortedByScore(users);
+            if(sortedUsers.Count >= 10)
```
Comment (aurisgo) [inline RoadRangerBackEnd/Score/HighscoresService.cs]: Use constant

File: RoadRangerBackEnd/Score/HighscoresService.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                }
+            }
+        }
+
+        public List<User> SortedByScore(List<User> users) //Isrykiuoja User lista pagal scora mazejimo tvarka
+        {
+            return users.OrderByDescending(o=>o.Score).ToList();
+        }
+
+        public List<User> Top10(List<User> users)
```
Comment (aurisgo) [inline RoadRangerBackEnd/Score/HighscoresService.cs]: It should depend on constant variable. So, rename it like GetTops.

File: RoadRangerBackEnd/Users/User.cs [hunk 5]
```diff
@@ -9,20 +9,21 @@ public class User : IComparable<User>
         public string Username { get; set; }
         public string Password { get; set; }
         public string Name { get; set; }
-        public int score = 0;
+        public int Score { get; set; }
         public Boosts boosts = new Boosts();
```
Comment (aurisgo) [inline RoadRangerBackEnd/Users/User.cs]: Use property

File: RoadRangerBackEnd/Score/HighscoresService.cs (hunk 3 above)
Comment (pauliusanuzas) [reply RoadRangerBackEnd/Score/HighscoresService.cs]: Done. :)

File: RoadRangerBackEnd/Score/HighscoresService.cs (hunk 4 above)
Comment (pauliusanuzas) [reply RoadRangerBackEnd/Score/HighscoresService.cs]: Done. :)

File: RoadRangerBackEnd/Users/User.cs (hunk 5 above)
Comment (pauliusanuzas) [reply RoadRangerBackEnd/Users/User.cs]: Done. :)

File: RoadRangerMobileApp/RoadRangerMobileApp/ViewModels/MyShopViewModel.cs (hunk 2 above)
Comment (pauliusanuzas) [reply RoadRangerMobileApp/RoadRangerMobileApp/ViewModels/MyShopViewModel.cs:13]: Will fix that when migrating the shop page to MVP structure.

File: RoadRangerMobileApp/RoadRangerMobileApp/Presenters/TopPresenter.cs (hunk 1 above)
Comment (pauliusanuzas) [reply RoadRangerMobileApp/RoadRangerMobileApp/Presenters/TopPresenter.cs]: Done. :)

**Pull request merged**